### PR TITLE
865 remove garmin activity schema

### PR DESCRIPTION
**Pull request by Goncharo**
Ref: metriport/metriport-internal#865

### Dependencies

N/A

### Description

remove garmin activity schema

### Release Plan

asap

### CODE REVIEW

File: api/app/src/mappings/garmin/activity-types.ts [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+
+export function activityTypeReadable(activityType: string): string {
+  // will the activity type mapping if it exists, and otherwise for example will change strings from:
+  //    - "SWIM_TO_BIKE_TRANSITION_V2" to "Swim To Bike Transition"
+  //    - "CROSS_COUNTRY_SKIING_WS" to "Cross Country Skiing"
+  const stringToUse = containsStringIndex(activityTypeMapping, activityType)
+    ? activityTypeMapping[activityType as ActivityType]
+    : activityType.replace(/_|V\d+|WS/g, " ");
+  return capitalize(stringToUse).trim();
+}
```
Comment (leite08) [inline api/app/src/mappings/garmin/activity-types.ts:16]: This is a great candidate for some unit tests! 🤓

File: api/app/src/mappings/garmin/activity-types.ts [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+
+// see https://github.com/metriport/metriport-internal/issues/865 for context...
+// these enums are current as of the Garmin API @ 1.0.3, and are out-of-date at the time of writing
+// this @ 1.0.5 - keeping for backwards compatibility
 export const activityTypeSchema = z.enum([
```
Comment (leite08) [inline api/app/src/mappings/garmin/activity-types.ts:21]: Where is this used now?

**Review by leite08 — APPROVED** [review]
LGTM, would just add some unit tests for diff situations on that function and, if possible, remove the schema.

File: api/app/src/mappings/garmin/activity-types.ts (hunk 2 above)
Comment (Goncharo) [reply api/app/src/mappings/garmin/activity-types.ts:21]: just to make the ActivityType enum https://github.com/metriport/metriport/blob/a4bb37cf978ff1b8848c6b96d3ce8fdd73897cf1/api/app/src/mappings/garmin/activity-types.ts#L123

Comment (Goncharo) [reply api/app/src/mappings/garmin/activity-types.ts:21]: could just change this to a <string, string> map though

File: api/app/src/mappings/garmin/activity-types.ts (hunk 1 above)
Comment (Goncharo) [reply api/app/src/mappings/garmin/activity-types.ts:16]: I had a feeling this was going to be a comment 😆 

Will add some tests

**Review by leite08 — APPROVED** [review]
haha, love the activities on the unit tests! :shipit:

**Review by leite08 — COMMENTED** [review]
Forgot to mention, we're not testing the cases where the conversion comes from the `activityTypeMapping`, especially the ones that are not based on the regex logic - maybe could add one of those, like `VIRTUAL_RUN: "VIRTUAL RUNNING"`

Also, just noticed what seems like a bug:

`SAILING_V2: "SAILING SAILING",`

File: api/app/src/mappings/garmin/__tests__/activity-types.test.ts [hunk 3]
```diff
@@ -13,6 +13,10 @@ describe("garmin activity types", () => {
     expect(activityTypeReadable("HEADBANGING_AT_CONCERT_V4")).toEqual("Headbanging at concert");
   });
 
+  test("unknown one-word activity type containing V2", async () => {
+    expect(activityTypeReadable("SAILING_V2")).toEqual("Sailing");
+  });
+
```
Comment (leite08) [inline api/app/src/mappings/garmin/__tests__/activity-types.test.ts:19]: nit: test name refers to "unknown", but `SAILING_V2` is known/mapped, right?

**Pull request merged**